### PR TITLE
Add modern address types and decorators to bip122/caip10

### PR DESCRIPTION
**Pull request by bumblefudge**
Best-effort at documenting without first-hand experience, open to input!

### CODE REVIEW

File: bip122/caip10.md [hunk 1]
```diff
@@ -1,30 +1,59 @@
...
+
+Rough validation for other networks can be achieved by updating the prefixes in the first enumeration of literals for the network-specific address-types targeted, and (if needed) updating the minimum and maximum lengths accordingly.
+
+For more precise validation, it is recommended to first detect the type from the prefix as mentioned above, then compute the type-specific checksum as specified in the corresponding Bitcoin Improvement Proposals linked above (taking into account that networks other than Bitcoin mainnet use different literals, affecting not just the prefixes but also the checksum algorithms).
+
+### Decorators
+
+In the Bitcoin development community, certain user experience conventions have arisen that can be opted into by applications and wallets, yet are, strictly speaking, not part of the bitcoin protocol and leave no on-chain artefacts in transaction history.
+By analogy to web standards, these are best designated with a matrix parameter (i.e., a `#parameter` suffix on the URI), which might be meaningful or useful to clients but are not native to the underlying URL protocol and are traditionally omitted when sent over the wire for resolution.
```
Comment (silverpill) [inline bip122/caip10.md:51]: `#parameter` in a URI is a [fragment](https://www.w3.org/DesignIssues/Fragment.html) identifier.
[Matrix](https://www.w3.org/DesignIssues/MatrixURIs.html) parameters are denoted by semicolons (and are not widely supported).

Comment (bumblefudge) [reply bip122/caip10.md:51]:
```suggestion
By analogy to web standards, these are best designated with a fragment identifier (i.e., a `#parameter` suffix on the URI), which might be meaningful or useful to clients but are not native to the underlying URL protocol and are traditionally omitted when sent over the wire for resolution.
```

File: bip122/caip10.md [hunk 2]
```diff
@@ -51,14 +99,18 @@ bip122:12a765e31ffd4059bada1e25190f6e98:128Lkh3S7CkDTBZ8W7BbpsN3YYizJMp8p6
...
 [BIP122]: https://github.com/bitcoin/bips/blob/master/bip-0122.mediawiki
-[CAIP-2]: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md
+[BIP141]: https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki
+[BIP173]: https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki
+[BIP350]: https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki
+[bitcoin whitepaper]: https://www.ussc.gov/sites/default/files/pdf/training/annual-national-training-seminar/2018/Emerging_Tech_Bitcoin_Crypto.pdf
 [CAIP-10]: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md
-[CAIP-19]: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-19.md
-[CAIP-21]: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-21.md
-[CAIP-22]: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-22.md
```
Comment (bumblefudge) [inline bip122/caip10.md:114]:
```suggestion
[RFC-1738]: https://www.rfc-editor.org/rfc/rfc1738

```

File: bip122/caip10.md [hunk 3]
```diff
@@ -1,30 +1,59 @@
...
+
+Rough validation for other networks can be achieved by updating the prefixes in the first enumeration of literals for the network-specific address-types targeted, and (if needed) updating the minimum and maximum lengths accordingly.
+
+For more precise validation, it is recommended to first detect the type from the prefix as mentioned above, then compute the type-specific checksum as specified in the corresponding Bitcoin Improvement Proposals linked above (taking into account that networks other than Bitcoin mainnet use different literals, affecting not just the prefixes but also the checksum algorithms).
+
+### Decorators
+
+In the Bitcoin development community, certain user experience conventions have arisen that can be opted into by applications and wallets, yet are, strictly speaking, not part of the bitcoin protocol and leave no on-chain artefacts in transaction history.
+By analogy to web standards, these are best designated with a fragment identifier (i.e., a `#parameter` suffix on the URI), which might be meaningful or useful to clients but are not native to the underlying URL protocol and are traditionally omitted when sent over the wire for resolution.
```
Comment (bumblefudge) [inline bip122/caip10.md:51]:
```suggestion
By analogy to web standards, these are best designated with "fragment identifier" of the sort specified in the [URL standard][RFC-1738] (i.e., a `#parameter` suffix on the URI), which might be meaningful or useful to clients but are not native to the underlying URL protocol and are traditionally omitted when sent over the wire for resolution.
```

**Review by GiMa-Maya — APPROVED** [review]

**Pull request merged**